guidance/library/_json.py

```diff
@@ -68,320 +235,333 @@ class Keyword(str, Enum):
+        self,
+        anyof_list: Sequence[Mapping[str, Any]],
+    ) -> GrammarNode:
+        options = [self._gen_json(json_schema=item) for item in anyof_list]
+        return self.builder.select(options)
+
+    def _process_enum(self, options: Sequence[Mapping[str, Any]]) -> GrammarNode:
+        # TODO: can we support a whitespace-flexible version of this?
+        return self.builder.select([_to_compact_json(opt) for opt in options])
+
+    def _gen_json_object(
+        self,
+        properties: Mapping[str, Any],
+        additional_properties: Union[bool, Mapping[str, Any]],
+    ) -> GrammarNode:
+        if additional_properties is True:
+            # True means that anything goes
+            additional_properties = {}
+
+        lm: List[Grammar] = ["{"]
+        if properties:
+            lm += self._process_properties(properties=properties)
+            if additional_properties is not False:
+                lm.append(",")
+        if additional_properties is not False:
+            lm.append(self._process_additional_properties(additional_properties))
+        lm.append("}")
+        return self.join(*lm)
+
+    def _process_properties(
+        self,
+        properties: Mapping[str, Any],
+    ) -> List[Grammar]:
+        res: List[Grammar] = []
+        properties_added = 0
+        for name, property_schema in properties.items():
+            res.append('"' + name + '"')
+            res.append(":")
+            res.append(self._gen_json(property_schema))
+            properties_added += 1
+            if properties_added < len(properties):
+                res.append(",")
+        return res
+
+    def join(self, *args: Grammar) -> GrammarNode:
+        return self.builder.join(args)
+
+    def _process_additional_properties(
+        self,
+        additional_properties: Mapping[str, Any],
+    ) -> GrammarNode:
+        item = self.join(self.json_simple_string, ":", self._gen_json(additional_properties))
+        return self.builder.optional(self._sequence(item))
+
+    def _sequence(self, item: GrammarNode) -> GrammarNode:
+        return self.builder.join([self.builder.zero_or_more(self.join(item, ",")), item])
+
+    @cached_property
+    def json_int(self) -> GrammarNode:
+        return self.builder.lexeme(r"-?(?:0|[1-9][0-9]*)")
+
+    @cached_property
+    def json_number(self) -> GrammarNode:
+        # TODO: guidance had a select() including json_int() here
+        return self.builder.lexeme(r"-?(?:0|[1-9][0-9]*)(?:\.[0-9]+)?(?:[eE][+-]?[0-9]+)")
+
+    @cached_property
+    def json_simple_string(self) -> GrammarNode:
+        return self.builder.lexeme('"' + CHAR_REGEX + "*" + '"')
+
+    def _gen_json_string(
+        self,
+        min_length: int = 0,
+        max_length: Union[int, None] = None,
+        regex: Optional[str] = None,
+    ) -> GrammarNode:
+        if min_length == 0 and max_length is None and regex is None:
+            return self.json_simple_string
+
+        if regex is not None:
```

Comment:
Something in here may not be quite correct....or perhaps it's in the copy-over that something needs updating.